docs/source/management/data-cleanup.md

```diff
@@ -0,0 +1,112 @@
+cleanup jobs, {{productName}} ensures the cluster is stable (conditions: `StatefulSetControllerProgressing=False`,
+`Available=True` and `Degraded=False`).
+
+:::{note}
+Due to {{productName}} relying strictly on the token ring changes, there are some limitations which we describe in the
+[Known limitations](#known-limitations) section.
+:::
+
+## Inspecting cleanup jobs
+
+{{productName}} creates a Job per node that requires cleanup. If any job is still running, the `ScyllaCluster`
+status contains the `JobControllerProgressing` condition set to `True`. When a job completes successfully, it's removed
+from the cluster by {{productName}}. The condition's message contains name of the running job(s).
+
+When no cleanup jobs are running, the `JobControllerProgressing` condition is set to `False`.
+
+You can inspect the condition by running:
+
+```shell
+kubectl get scyllacluster <sc-name> -o jsonpath='{.status.conditions[?(@.type=="JobControllerProgressing")]}' | jq
+```
+
+Where `<sc-name>` is the name of your ScyllaCluster.
+
+You should see output similar to this (when no jobs are running):
+
+```json
+{
+  "lastTransitionTime": "2025-12-12T13:51:52Z",
+  "message": "",
+  "observedGeneration": 2,
+  "reason": "AsExpected",
+  "status": "False",
+  "type": "JobControllerProgressing"
+}
+```
+
+:::{caution}
+You may not see the cleanup jobs in the cluster if they complete quickly (e.g., on small datasets or cluster bootstrap)
+and are removed before you can inspect them.
+
+To ensure they were created and completed, you can inspect Kubernetes events:
+
+```shell
+kubectl get events | grep job
+```
+
+The output should contain entries similar to:
+
+```shell
+30m  Normal  JobCreated        job/cleanup-scylla-us-east-1-us-east-1a-0  Job default/cleanup-scylla-us-east-1-us-east-1a-0 created
+30m  Normal  SuccessfulCreate  job/cleanup-scylla-us-east-1-us-east-1a-0  Created pod: cleanup-scylla-us-east-1-us-east-1a-0-tpd7x
+30m  Normal  Completed         job/cleanup-scylla-us-east-1-us-east-1a-0  Job completed
+30m  Normal  JobDeleted        job/cleanup-scylla-us-east-1-us-east-1a-0  Job default/cleanup-scylla-us-east-1-us-east-1a-0 deleted
+```
+
+You can see that for each cleanup job, there are events for job creation, pod creation, job completion, and job deletion.
+:::
+
+## Known limitations
+
+{{productName}} triggers cleanup based on token ring changes. While this approach is safe, it has some side effects where
+cleanup may be triggered unnecessarily or not at all.
+
+### Unnecessary cleanups on node decommission
+
+When a node is decommissioned, tokens are redistributed to other remaining nodes. These nodes do not require cleanup since
+they are not losing any tokens. However, {{productName}} still triggers cleanup for these nodes due to the token ring change.
+This is a safe operation, but it may lead to unnecessary spike in I/O load.
```

Comment:
```suggestion
When you decommission a node, tokens are redistributed to the remaining nodes. These nodes do not require cleanup since
they are not losing any tokens. However, {{productName}} still triggers cleanup for these nodes due to the token ring change.
This is a safe operation, but it may lead to an unnecessary spike in I/O load.
```
nit
per style guide + a small error